lib/Conversion/TorchToMhlo/BasicOp.cpp

```diff
@@ -25,6 +26,470 @@ using namespace mlir;
 using namespace mlir::torch;
 using namespace mlir::torch::Torch;
 
+bool skipMultiplyAlpha(Value alphaValue) {
+  double doubleValue;
+  auto isFloat = matchPattern(alphaValue, m_TorchConstantFloat(&doubleValue));
+
+  int64_t intValue;
+  auto isInt = matchPattern(alphaValue, m_TorchConstantInt(&intValue));
+
+  return ((isFloat && doubleValue == 1.0) || (isInt && intValue == 1.0));
+}
+
+// These legalizations are for unary ops with only for floating point datatypes.
+// There is no supported quantized integer mode for these.
+namespace {
+template <typename AtenOpT, typename MhloOpT>
+class ConvertAtenUnaryFPOnlyOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value self = adaptor.self();
+    auto selfTy = self.getType().cast<TensorType>();
+
+    if (!selfTy)
+      return op.emitError("Only Tensor types supported in MHLO");
+
+    if (selfTy.getElementType().isa<mlir::FloatType>()) {
+      rewriter.replaceOpWithNewOp<MhloOpT>(
+          op,
+          OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+              op.getType()),
+          self);
+      return success();
+    } else {
+      return op.emitError(
+          "Only floating-point datatype legalization supported");
+    }
+  }
+};
+} // namespace
+
+// aten.ones & aten.zeros
+// Ref: Error checking based on the Torch to TOSA lowering
```

Comment:
Nit: TOSA -> MHLO

Reply:
This part's implementation largely borrows from Torch-to-TOSA lowering. So we marked "TOSA" here.